reviewrot/gitlabstack.py

```diff
@@ -0,0 +1,170 @@
+        If repo name is not provided then requests merge requests
+        for all repos for specified username/organization.
+
+        Args:
+            user_name (str): Gitlab namespace
+            repo_name (str): Gitlab project name for specified
+                          namespace
+            state_ (str): The state for pull requests, e.g, older
+                        or newer
+            value (int): The value in terms of duration for requests
+                         to be older or newer than
+            duration (str): The duration in terms of period(year, month,
+                            hour, minute) for requests to be older or
+                            newer than
+            token (str): Gitlab token for authentication
+            host (str): Gitlab host name for authentication
+
+        Returns:
+            response (list): Returns list of list of pull requests for
+                             specified user(group) name and projectname or all
+                             projectname for given groupname
+        """
+        gl = gitlab.Gitlab(host, token)
+        gl.auth()
+        log.debug('Gitlab instance created: %s', gl)
+        response = []
+        # if Repository name is explicitely provided
+        if repo_name is not None:
+            try:
+                # get project object for given repo_name(project name)
+                project = gl.projects.get(os.path.join(user_name, repo_name))
+            except GitlabGetError:
+                log.exception('Project %s not found for user %s',
+                              repo_name, user_name)
+                raise Exception('Project %s not found for user %s'
+                                % (repo_name, user_name))
+            # get merge requests for specified username and project name
+            res = self.get_reviews(uname=user_name, project=project,
+                                   state_=state_, value=value,
+                                   duration=duration)
+
+        else:
+            # get user object
+            groups = gl.groups.search(user_name)
+            if not groups:
+                log.debug('Invalid user/group name: %s', user_name)
+                raise Exception('Invalid user/group name: %s' % user_name)
+
+            # get merge requests for all projects for specified group
+            for group in groups:
+                projects = gl.group_projects.list(group_id=group.id)
+                if not projects:
+                    log.debug("No projects found for user/group name %s",
+                              user_name)
+                for project in projects:
+                    res = self.get_reviews(uname=user_name, project=project,
+                                           state_=state_, value=value,
+                                           duration=duration)
+                    response.append(res)
+        return response
+
+    def get_reviews(self, uname, project, state_=None,
```

Comment:
Almost everything in this method is identical to what you have in the `GithubService` class. It is also repeated in `PagureService` too. Can you abstract it into the base class?